starlark/src/syntax/ast.rs

```diff
@@ -335,6 +337,60 @@ impl Statement {
         }
         Ok(Statement::Def(name, parameters, stmts))
     }
+
+    /// Validate `break` and `continue` is only used inside loops
+    fn validate_break_continue(stmt: &AstStatement) -> Result<(), Diagnostic> {
+        match stmt.node {
+            Statement::Break | Statement::Continue => {
+                let kw = if let Statement::Break = stmt.node {
+                    "break"
+                } else {
+                    "continue"
+                };
+                Err(Diagnostic {
+                    level: Level::Error,
+                    message: format!("{} cannot be used outside of loop", kw),
+                    code: Some("CS08".to_owned()),
```

Comment:
Extract the error code like for the rest of the module.

BTW if you have an idea on how to organize those error codes, feel free to share it.

Reply:
> Extract the error code like for the rest of the module.

In the rest of the module error codes are defined inline, but it's good idea to make constants for them. I'll do it.

> if you have an idea on how to organize those error codes, feel free to share it.

I'd do it like this: make a single big module with error code. Allocate two prefixes: `CT` for compile-time errors (like parser errors or break outside of loop) and `RT` for runtime errors like index out of range.